plugins/Demo/DateSetting/DateSetting.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * DateSetting Plugin for LimeSurvey
+ * @author : Denis Chenu
+ * @copyright: LimeSurvey <https://community.limesurvey.org/>
+ * @version:; 0.1.0
+ * @licence MIT Licence
+ */
+class TestDateSetting extends PluginBase
+{
+    /** @inheritdoc **/
+    protected static $name = 'TestDateSetting';
+    /** @inheritdoc **/
+    protected static $description = 'Pluginto test HTML DB setting.';
+    /** @inheritdoc **/
+    protected $storage = 'DbStorage';
+    /** @inheritdoc **/
+    protected $settings = [
+        'checkDateDate' => array(
+            'type' => 'date',
+            'label' => 'A date only saved',
+            'saveformat' => "Y-m-d",
+        ),
+        'checkDateYear' => array(
+            'type' => 'date',
+            'label' => 'A year only saved',
+            'saveformat' => "Y",
+        ),
+        'checkDateDefault' => array(
+            'type' => 'date',
+            'label' => 'Another date no format set',
+        ),
+        'checkDateFalse' => array(
+            'type' => 'date',
+            'label' => 'A date saved as shown',
+            'saveformat' => false,
+        ),
+    ];
+
+    /** @inheritdoc **/
+    public function init()
+    {
+        $this->subscribe('beforeSurveySettings');
+        $this->subscribe('newSurveySettings');
+    }
+
+    /**
+     * Register the settings
+     */
+    public function beforeSurveySettings()
+    {
+        $oEvent = $this->event;
+        $surveyId = $oEvent->get('survey');
+        $oEvent->set(
+            "surveysettings.{$this->id}",
+            array(
+                'name' => get_class($this),
+                'settings' => array(
+                    'checkSurveyDateDate' => array(
+                        'type' => 'date',
+                        'label' => 'A date only saved',
```

Comment:
Not needed, but I think we could show on helptext the value stored on the DB, as to show the difference in between how is selected and how is saved.